yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -372,11 +381,29 @@ private[spark] class Client(
       sparkConf.set(CONF_SPARK_YARN_SECONDARY_JARS, cachedSecondaryJarLinks.mkString(","))
     }
 
+    if (args.primaryPyFile != null) {
+      distribute(args.primaryPyFile)
```

Comment:
i think primaryPyFile is only used in applicationMaster, so it is distributed to appMaster.